Reference_Manual.md

```diff
@@ -1269,22 +1273,70 @@ The following is an example configuration where we set the secret token bytes to
 
 ```c++
 std::vector<hazelcast::byte> my_token = {'S', 'G', 'F', '6', 'Z', 'W'};
+config.set_credentials(std::make_shared<hazelcast::client::security::token_credentials>(my_token));
+auto hz = hazelcast::new_client(std::move(config)).get();
+```
+Implementing a custom login module is mandatory to use token authentication by itself. An example custom login module is:
+```Java
+import com.hazelcast.config.Config;
+import com.hazelcast.security.ClusterLoginModule;
+import com.hazelcast.security.Credentials;
+import com.hazelcast.security.CredentialsCallback;
+import com.hazelcast.security.TokenCredentials;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class TokenLoginModule extends ClusterLoginModule {
+
+    private String name;
+    private final byte[] expectedToken = "SGF6ZW".getBytes(StandardCharsets.US_ASCII);
 
-hazelcast_client hz(client_config().set_credentials(
-std::make_shared<security::token_credentials>(my_token)));
+    @Override
+    protected boolean onLogin() throws LoginException {
+        CredentialsCallback cb = new CredentialsCallback();
+	name = cb.getCredentials().getName();
```

Comment:
minor: Let's move this line below `Credentials credentials = cb.getCredentials();`, you already call `getCredentials` there, no need to duplicate it.